// agentmail/dist/esm/serialization/resources/organizations/types/Organization.mjs
// This file was auto-generated by Fern from our API Definition.
import * as core from "../../../../core/index.mjs";
import { OrganizationId } from "../../../types/OrganizationId.mjs";
export const Organization = core.serialization.object({
    organizationId: core.serialization.property("organization_id", OrganizationId),
    inboxCount: core.serialization.property("inbox_count", core.serialization.number()),
    domainCount: core.serialization.property("domain_count", core.serialization.number()),
    inboxLimit: core.serialization.property("inbox_limit", core.serialization.number().optional()),
    domainLimit: core.serialization.property("domain_limit", core.serialization.number().optional()),
    billingId: core.serialization.property("billing_id", core.serialization.string().optional()),
    billingType: core.serialization.property("billing_type", core.serialization.string().optional()),
    billingSubscriptionId: core.serialization.property("billing_subscription_id", core.serialization.string().optional()),
    authenticationId: core.serialization.property("authentication_id", core.serialization.string().optional()),
    authenticationType: core.serialization.property("authentication_type", core.serialization.string().optional()),
    updatedAt: core.serialization.property("updated_at", core.serialization.date()),
    createdAt: core.serialization.property("created_at", core.serialization.date()),
});
